tests/language-feature/interfaces/generic-requirement-synth.slang

```diff
@@ -0,0 +1,90 @@
+// Test that we can synthesize requirements for generic methods.
+
+//TEST(compute):COMPARE_COMPUTE_EX(filecheck-buffer=CHECK):-dx11 -compute  -output-using-type
+//TEST(compute):COMPARE_COMPUTE_EX(filecheck-buffer=CHECK):-vk -compute  -output-using-type
+
+interface IBase
+{
+    static float get();
+}
+interface IBar : IBase
+{
+    float derivedMethod();
+}
+
+struct Bar : IBar
+{
+    static float get() { return 1.0f; }
+    float derivedMethod() { return 2.0f; }
+}
+
+interface ITestInterface<Real : IFloat>
+{
+    Real sample<T : IBar>(T t);
```

Comment:
I was curious if it would work when I have another method like `sample<T:IBar2>(T t)` where IBar2 inherits from IBase too.
It seems to be working as expected.
```
interface IBar2 : IBase
{
    float derivedMethod2();
}

struct Bar2 : IBar2
{
    static float get() { return 1.0f; }
    float derivedMethod2() { return 3.0f; }
}

interface ITestInterface<Real : IFloat>
{
    Real sample<T : IBar>(T t);
    Real sample<T : IBar2>(T t); // <=== This seems to be synthesized as well.
```

Reply:
Yes, in this case, both requirements can be synthesized from a single implementation that is `Real sample<T:IBae>()`.